### PR TITLE
Upgrade hibernate to 5.6.5.Final and fix compilation errors

### PR DESCRIPTION
**Pull request by mpeddada1**
Running `./mvnw verify -f google-cloud-spanner-hibernate-testing/pom.xml -Dhibernate.show_sql=false -Dhibernate.connection.url='jdbc:cloudspanner:/projects/cloud-spanner-hibernate-ci/instances/test-instance/databases/test-database'`  locally results in a successful build.

### CODE REVIEW

File: google-cloud-spanner-hibernate-dialect/src/test/java/com/google/cloud/spanner/hibernate/GeneratedUpdateTableStatementsTests.java
Comment (mpeddada1) [inline google-cloud-spanner-hibernate-dialect/src/test/java/com/google/cloud/spanner/hibernate/GeneratedUpdateTableStatementsTests.java:97]: Hibernate 5.5.2.final is resulting in some schema changes when updating a table that already exists.

File: google-cloud-spanner-hibernate-dialect/src/test/java/com/google/cloud/spanner/hibernate/SpannerForeignKeyExporterTests.java
Comment (mpeddada1) [inline google-cloud-spanner-hibernate-dialect/src/test/java/com/google/cloud/spanner/hibernate/SpannerForeignKeyExporterTests.java:72]: Fixes test that was resulting `"getSqlDropStrings() was not implemented!"` error message which was caused due to deprecation of `getSqlDropStrings(T exportable, Metadata metadata)`

File: google-cloud-spanner-hibernate-dialect/src/test/java/com/google/cloud/spanner/hibernate/SpannerTableExporterTests.java
Comment (mpeddada1) [inline google-cloud-spanner-hibernate-dialect/src/test/java/com/google/cloud/spanner/hibernate/SpannerTableExporterTests.java:142]: Hibernate now uses the `;` separator as the default when generating DDL scripts: https://in.relation.to/2021/06/15/hibernate-orm-552-release/

**Comment by husseyd** [comment]
Checks passing - anything holding up a new release?

File: google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/schema/SpannerForeignKeyExporter.java
Comment (suztomo) [inline google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/schema/SpannerForeignKeyExporter.java:44]: Memo: a backward incompatible change. Next version should have a major version bump (unless this class is not meant for users).

File: google-cloud-spanner-hibernate-dialect/src/test/java/com/google/cloud/spanner/hibernate/GeneratedUpdateTableStatementsTests.java
Comment (suztomo) [reply google-cloud-spanner-hibernate-dialect/src/test/java/com/google/cloud/spanner/hibernate/GeneratedUpdateTableStatementsTests.java:97]: Would you describe the problem a bit more?

These deleted lines look as if hibernate 5.6.5.Final doesn't work with these alter table statements.

Comment (mpeddada1) [reply google-cloud-spanner-hibernate-dialect/src/test/java/com/google/cloud/spanner/hibernate/GeneratedUpdateTableStatementsTests.java:97]: Experimented with this a bit. Versions 5.4.6 and 5.5.0.final resulted in no change in schema, however versions 5.5.2.final onwards appear to drop the command that adds columns to a pre-existing table. This looks like a schema update from hibernate orm. Optimizations tend to be put in place with every [version upgrade ](https://in.relation.to/2021/06/15/hibernate-orm-552-release/) so I wonder if this is related. Also found a similar issue that is related to updating existing tables: https://github.com/hibernate/hibernate-orm/pull/3151

Comment (suztomo) [reply google-cloud-spanner-hibernate-dialect/src/test/java/com/google/cloud/spanner/hibernate/GeneratedUpdateTableStatementsTests.java:97]: I see. Would you add source code comment `// Why not seeing alter table here? It's because in hibernate 5.5.2 onwards ...`

**Review by suztomo — APPROVED** [review]
Approved with a suggestion for source code comment.

**Review by meltsufin — APPROVED** [review]

File: google-cloud-spanner-hibernate-dialect/src/test/java/com/google/cloud/spanner/hibernate/GeneratedUpdateTableStatementsTests.java
Comment (mpeddada1) [reply google-cloud-spanner-hibernate-dialect/src/test/java/com/google/cloud/spanner/hibernate/GeneratedUpdateTableStatementsTests.java:97]: Good idea! It'll be helpful to mention this.

File: google-cloud-spanner-hibernate-dialect/src/test/java/com/google/cloud/spanner/hibernate/GeneratedUpdateTableStatementsTests.java
Comment (suztomo) [inline google-cloud-spanner-hibernate-dialect/src/test/java/com/google/cloud/spanner/hibernate/GeneratedUpdateTableStatementsTests.java:96]: @mpeddada1 Thanks. It might be helpful to give some feedback to Hibernate library maintainers if this is undesirable/unexpected behavior of the library from your perspective.

**Pull request merged**
